orttraining/orttraining/python/training/ortmodule/_custom_autograd_function.py

```diff
@@ -25,13 +25,17 @@ def enable_custom_autograd_support():
     from torch.onnx import register_custom_op_symbolic
     from ._custom_autograd_function_exporter import _export
     from ._custom_autograd_function_runner import call_python_forward_function, call_python_backward_function
+    from onnxruntime.training.ortmodule.torch_cpp_extensions import torch_interop_utils
     import atexit
 
     register_forward_runner(call_python_forward_function)
     register_backward_runner(call_python_backward_function)
 
     # Unregister all python functions automatically upon normal interpreter termination.
     atexit.register(unregister_python_functions)
+    # Clear all gradient functions, to avoid a deadlock issue.
+    # Check the called function for more detailed comments.
+    atexit.register(torch_interop_utils.clear_all_grad_fns)
```

Comment:
Oh yeah, you are right, we can move this into torch_interop_utils module. 

For the second question, I did not find such an approach acting as C++ destructors corresponding to __init__ constructor. 
In terms of RAII, do you mean we still add python context covering ORTModule forward + backward run? I believe this is not feasible? Feel free to correct me if I am wrong.

Reply:
No worries. I was just hoping that we could somehow auto destruct the object based on some signal. Seems like it might be not straightforward. I think this is good for now.